core/geometry/src/curve/internalContexts/EllipticalArcApproximationContext.ts

```diff
@@ -0,0 +1,1092 @@
+    return this._remapFunction;
+  }
+  public set remapFunction(f: FractionMapper) {
+    this._remapFunction = f;
+  }
+};
+
+/**
+ * Structured data carrier used by the elliptical arc sampler.
+ * @internal
+*/
+export class QuadrantFractions {
+  private _quadrant: 1 | 2 | 3 | 4;
+  private _fractions: number[];
+  private _axisAtStart: boolean;
+  private _axisAtEnd: boolean;
+  private _averageAdded: boolean;
+  private constructor(
+    quadrant: 1 | 2 | 3 | 4, fractions: number[], axisAtStart: boolean = false, axisAtEnd: boolean = false,
+  ) {
+    this._quadrant = quadrant;
+    this._fractions = fractions;
+    this._axisAtStart = axisAtStart;
+    this._axisAtEnd = axisAtEnd;
+    this._averageAdded = false;
+  }
+  /** Constructor, captures the array. */
+  public static create(
+    quadrant: 1 | 2 | 3 | 4, fractions: number[] = [], axisAtStart: boolean = false, axisAtEnd: boolean = false,
+  ): QuadrantFractions {
+    return new QuadrantFractions(quadrant, fractions, axisAtStart, axisAtEnd);
+  }
+  /**
+   * Quadrant of the full ellipse containing the samples.
+   * * Quadrants are labeled proceeding in counterclockwise angular sweeps of length pi/2 starting at vector0.
+   * * For example, Quadrant 1 starts at vector0 and ends at vector90, and Quadrant 4 ends at vector0.
+   * * For purposes of angle classification, quadrants are half-open intervals, closed at their start angle,
+   * as determined by the ellipse's sweep direction.
+   */
+  public get quadrant(): 1 | 2 | 3 | 4 {
+    return this._quadrant;
+  }
+  /** Sample locations in this quadrant of the elliptical arc, as fractions of its sweep. */
+  public get fractions(): number[] {
+    return this._fractions;
+  }
+  public set fractions(f: number[]) {
+    this._fractions = f;
+  }
+  /**
+   * Whether the first fraction is the location of an ellipse axis point.
+   * * Only valid if `this.fractions.length > 0`.
+   */
+  public get axisAtStart(): boolean {
+    return this._fractions.length > 0 ? this._axisAtStart : false;
+  }
+  public set axisAtStart(onAxis: boolean) {
+    this._axisAtStart = onAxis;
+  }
+  /**
+   * Whether the last fraction is the location of an ellipse axis point.
+   * * Only valid if `this.fractions.length > 1`.
+   */
+  public get axisAtEnd(): boolean {
+    return this._fractions.length > 1 ? this._axisAtEnd : false;
+  }
+  public set axisAtEnd(onAxis: boolean) {
+    this._axisAtEnd = onAxis;
+  }
+  /**
+   * Whether the average fraction was added to produce a fractions array of length 3.
+   * * Some sample methods and elliptical arcs may not yield enough samples in a quadrant.
+   * * This flag is set if an extra (average) fraction was added to a 2-fraction array to satisfy
+   * the minimum 3-sample requirement that allows both axis/endpoints in the quadrant to be interpolated
+   * by [[EllipticalArcApproximationContext.constructCircularArcChainApproximation]].
+  */
+  public get averageAdded(): boolean {
```

Comment:
ok so we need at least 3 fractions on each quadrant to be able to pass a circle from those 3 points. If we have 2 fractions `f1` and `f2` we add the third fraction of `(f1+f2)/2` and set this flag.

That first line of the comment is confusing. I had to read next lines multiple times to understand it. Also the name `average` is confusing. I would go with `extraFraction` instead.

I also update the comment to something like this:

Whether the extra fraction was added.
To generate a circle, we need 3 points but some sample methods and elliptical arcs may not yield enough samples in a quadrant. 
This flag is set if an extra (average) fraction was added to a 2-fraction array to satisfy the minimum 3-sample requirement that allows both axis/endpoints in the quadrant to be interpolated by [[EllipticalArcApproximationContext.constructCircularArcChainApproximation]].